drivers/gpio/gpio_sx1509b.c

```diff
@@ -348,6 +537,43 @@ static int sx1509b_init(struct device *dev)
 		goto out;
 	}
 
+#ifdef CONFIG_GPIO_SX1509B_INTERRUPT
+	drv_data->gpio_int = device_get_binding(cfg->gpio_int_dev_name);
+	if (!drv_data->gpio_int) {
+		/* interrupt not supported */
+		drv_data->interrupt_sup = 0;
```

Comment:
This branch should be:

    rc = -ENOTSUP;
    goto out;

because if interrupt configuration is enabled the binding has to be present, and if it's present and the pin doesn't exist the configuration is wrong and the driver should fail.

Ideally we wouldn't have `CONFIG_GPIO_SX1509B_INTERRUPT`, and would instead use the existence of the`nint-gpios` devicetree property.

Then you can get rid of the `interrupt_sup` flag (which should have been `bool` not `u8_t`).

Reply:
I agree. Using nint-gpios property automatically is better. I will try that next. Right now, I have retained CONFIG_GPIO_SX1509B_INTERRUPT. Thanks.